frontend/src/app/spinner.interceptor.ts

```diff
@@ -0,0 +1,27 @@
+import { Injectable } from '@angular/core';
+import {
+	HttpRequest,
+	HttpHandler,
+	HttpEvent,
+	HttpInterceptor
+} from '@angular/common/http';
+import { Observable } from 'rxjs';
+import { finalize } from 'rxjs/operators';
+import { SpinnerService } from './todo/list/spinner.service';
+
+@Injectable()
+export class SpinnerInterceptor implements HttpInterceptor
+{
+	constructor(private listService : SpinnerService)
+	{}
+
+	intercept(request : HttpRequest<unknown>, next : HttpHandler) : Observable<HttpEvent<unknown>>
+	{
+  	this.listService.show();
+  	return next.handle(request).pipe(
+		finalize(() =>
```

Comment:
This must have return value like this because it's Observable<HttpEvent<unknown>>. I try to separate the code and change it but the spinner is circling continuously and not hiding.